### PR TITLE
Updated the autogen header feature

### PR DESCRIPTION
**Pull request by jisonZ**
The tool is now capable of automatically generate ResourceAccessPoint.

### CODE REVIEW

File: INSTRUCTIONS.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+For this extra credit assignment, you will generate visualizations of Rust programs by annotating code according to the specifications found in the README file.
```
Comment (cyrus-) [inline INSTRUCTIONS.md]: Discussion about the extra credit assignment shouldn't be in the main repo.

File: README.md [hunk 2]
```diff
@@ -61,9 +60,11 @@ In [main.rs](src/examples/string_from_move_print/main.rs), we define these RAPs
 ```rust
 /*--- BEGIN Variable Definitions ---
 Owner x; Owner y;
-Function String::from()
+Function String::from();
 --- END Variable Definitions ---*/
 ```
+Notice that the definition header now could be automatically generated by running the [view_examples.sh](rustviz_mdbook/view_examples.sh) once. If any incorrect information appeared at the generated header, you could manully edit it by refering to the following documentation.
```
Comment (cyrus-) [inline README.md]: Remove "Notice that", change "could" -> "can" (present tense, not past, in things like this)

File: RustvizParse/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "parse"
+version = "0.1.0"
+authors = ["Haochen Zhang <haochenz@Haochens-MacBook-Pro.local>"]
```
Comment (cyrus-) [inline RustvizParse/Cargo.toml]: add at least myself to the authors list (and any of the other students that have contributed to this code), and fix the email address, also the "edition" shouldn't be "2018"

**Review by cyrus- — CHANGES_REQUESTED** [review]

**Review by cyrus- — APPROVED** [review]
Didn't review the code in detail, but the README delta looks good now. Ready to merge when you are.

**Review by jesssleepy — APPROVED** [review]
Except that the recursive check of function calls in macros misses some new functions, the other functions I've tested work well.

**Pull request merged**